### PR TITLE
initrd.systemd: disable emergency mode

### PR DESCRIPTION
**Pull request by mrVanDalo**
we disable emergency mode in systemd, but if systemd is enabled during boot we still end up in emergency mode eventually, this will fix that.

### CODE REVIEW

File: nixos/server/default.nix [hunk 1]
```diff
@@ -62,6 +62,14 @@
   # No mutable users by default
   users.mutableUsers = false;
 
+  # Given that our systems are headless, emergency mode is useless.
+  # We prefer the system to attempt to continue booting so
+  # that we can hopefully still access it remotely.
+  boot.initrd.systemd.suppressedUnits = [
```
Comment (Mic92) [inline nixos/server/default.nix:68]:
```suggestion
  boot.initrd.systemd.suppressedUnits = lib.mkIf config.systemd.enableEmergencyMode [
```

Comment (Mic92) [reply nixos/server/default.nix:68]: Users might still enable emergency mode in their configuration.

**Comment by mrVanDalo** [comment]
I cleaned up a bit this commits and rebase to **main**

**Comment by mrVanDalo** [comment]
Ok now everything is green.

**Comment by zimbatm** [comment]
thanks!

**Pull request merged**